src/utils/extractTextStyles.ts

```diff
@@ -0,0 +1,51 @@
+import type { TextStyle } from 'react-native';
+import { memoize } from './lodash';
+
+const textStyleProps = [
+    'color',
+    'fontFamily',
+    'fontSize',
+    'fontStyle',
+    'fontWeight',
+    'letterSpacing',
+    'lineHeight',
+    'textAlign',
+    'textDecorationLine',
+    'textDecorationStyle',
+    'textDecorationColor',
+    'textShadowColor',
+    'textShadowOffset',
+    'textShadowRadius',
+    'textTransform',
+] as const;
+
+const defaultTextStyle: TextStyle = {};
+const extract = <T extends TextStyle>(styles: T) => {
+    const { textStyle, otherStyle } = textStyleProps.reduce(
+        (obj, key) => {
+            if (key in obj.otherStyle) {
+                const value = obj.otherStyle[key] as T[typeof key];
+
+                // delete from the object;
+                obj.otherStyle = { ...obj.otherStyle };
+                delete obj.otherStyle[key];
```

Comment:
As I understood, delete isn't very performant for v8

Ref: https://github.com/googleapis/google-api-nodejs-client/issues/375

We can use lodash.omit here. It's already in utils file

Reply:
Benchmarking
https://www.measurethat.net/Benchmarks/Show/6301/2/lodash-omit-vs-native-delete---mantz#latest_results_block

Reply:
I do not know how you read the results.. but 3.4M operations on native is more than twice of 1.6M operations with lodash.omit :D

Reply:
It's not operations, it's execution per seconds.

Please also read the github issue above.

Reply:
It you think it's not that big of a deal. It's fine with me :)

Reply:
to be read in ops/second instead of seconds/operation